src/storage/db_interface_common.py

```diff
@@ -310,3 +311,50 @@ def release_unpacking_lock(self, uid):
 
     def drop_unpacking_locks(self):
         self.main.drop_collection('locks')
+
+    def _collect_analysis_tags_from_children(self, uid):
+        children = self._fetch_children_with_tags(uid)
+        unique_tags = {}
+        for child in children:
+            self._collect_analysis_tags(child, unique_tags)
+        return unique_tags
+
+    def _collect_analysis_tags(self, file_object, analysis_tags):
+        for name, analysis in [(n, a) for n, a in file_object.processed_analysis.items() if 'tags' in a]:
+            if 'file_system_flag' in analysis and analysis['file_system_flag']:
+                analysis = self.retrieve_analysis(file_object.processed_analysis, analysis_filter=[name, ])[name]
+
+            for tag_type, tag in analysis['tags'].items():
+                if tag_type != 'root_uid' and tag['propagate']:
+                    append_unique_tag(analysis_tags, tag, name, tag_type)
+
+    def _fetch_children_with_tags(self, uid):
+        plugins_with_tag_propagation = [  # FIXME This should be inferred in a sensible way. This is not possible yet.
+            'crypto_material', 'cve_lookup', 'known_vulnerabilities', 'qemu_exec', 'software_components',
+            'users_and_passwords'
+        ]
+        uids = set()
+        for plugin in plugins_with_tag_propagation:
+            uids.update(
+                set(
+                    get_list_of_all_values(
+                        self.file_objects,
+                        '$_id',
+                        match={
+                            'virtual_file_path.{}'.format(uid): {'$exists': 'true'},
+                            f'processed_analysis.{plugin}.tags': {'$exists': 'true'}
+                        }
+                    )
+                )
+            )
+        return self.get_objects_by_uid_list(uids, analysis_filter=plugins_with_tag_propagation)
```

Comment:
If at some point the performance should prove to be a problem, we should think about a solution with fewer DB queries. The performance seems to be fine, though

Reply:
Agreed